pkg/storage/test/tuples.go

```diff
@@ -317,6 +318,98 @@ func TupleWritingAndReadingTest(t *testing.T, datastore storage.OpenFGADatastore
 		_, err = gotTuples.Next()
 		require.ErrorIs(t, err, storage.ErrIteratorDone)
 	})
+
+	t.Run("reading_userset_tuples_with_filter_made_of_direct_relation_reference", func(t *testing.T) {
```

Comment:
Would you mind just extending this test slightly so that it returns filtered iterator with more than one type restriction?

e.g. add one more tuple

```
document:1#viewer@team:tigers#member
```
and then
```
ReadUsersetTuples(ctx, ..., storage.ReadUsersetTuplesFilter {
    Object: "document:1",
    Relation: "viewer",
    AllowedUserTypeRestrictions: []*openfgapb.RelationReference{
        typesystem.DirectRelationReference("group", "member"),
        typesystem.DirectRelationReference("team", "member"),
    },
})
```
This will give us confidence that our SQL `IN` or union works as we expect.